go/log_messages.md

```diff
@@ -0,0 +1,90 @@
+## Log Messages
+
+### Resource Reconciliation
+
+When doing resource reconciliation action we often have to find out the current
+state, and apply changes to bring it towards desired state. This often means
+calling some API and then emitting debug logs about that. We should try to
+align this logs to become more comfortable when reading them and find problems
+faster. Also we should aim for them to be understandable not only by developers
+but also by SREs, SEs and any other people who look at them and not
+necessarily contribute to the codebase.
+
+Usually the reconciling code discovers some facts about the system in order to
+figure out what actions it needs to take in order to push the current state
+towards the desired state. Those discovering blocks should look like this
+whenever possible (the additional comments should be omitted they serve
+informational purpose only):
+
+```go
+var err error
+
+// Note all messages contain "system fact" which is human readable name for the
+// system fact being discovered. Exception is "cancelling resource" message.
+var systemFact string
+{
+	// Start message. Note "finding out" here.
+	r.logger.LogCtx(ctx, "level", "debug", "message", "finding out system fact")
+
+	systemFact, err = pkg.FindSystemFact(ctx)
+	if IsNotFound(err) {
+		// Not found log. Note "did not find" here.
+		r.logger.LogCtx(ctx, "level", "debug", "message", "did not find system fact")
+		// Reason log should be separate free form log explaining why
+                // the system fact can't be found.
+		r.logger.LogCtx(ctx, "level", "debug", "message", "waiting for system fact to be created")
```

Comment:
Added a reason block which serves as an explanation for for the dry log of not being able to find info about the fact.

Reply:
Nitpick: There is an extra space in line 34.